lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1096,9 +1100,11 @@ instance ToJSON ApiByronWalletBalance where
     toJSON = genericToJSON defaultRecordTypeOptions
 
 instance FromJSON (ApiT PoolId) where
-    parseJSON = parseJSON >=> eitherToParser . bimap ShowFmt ApiT . fromText
+    parseJSON j = legacyParse j <|> (parseJSON >=> eitherToParser . bimap (ShowFmt . displayException) ApiT . decodePoolIdBech32 $ j)
```

Comment:
Note that the `FromJSON` instance is only really used for testing here, there's no API endpoint which expects a pool id in a JSON body. So the backward compatibility here is unnecessary. 

Users may however provide pool ids as part of the path parameters relying on `FromHttpApiData`; this needs to be updated to accept bech32 while indeed keeping support for base16.